rdkcentral/Lightning-UI-Components
# fix: allow componentConfig props to properly merge with component props

## Description

Props set in componentConfig are not properly merged and are leaving some style properties when themes are changed at runtime. Also any componentConfig properties should deep merge with properties that are set at the instance level.

## References

[Ticket](https://ccp.sys.comcast.net/browse/LUI-1217)

## Testing

1. Open the Tile component
2. Right click on the iframe window containing the component in storybook and open the webmaster tools
3. Observe that no image scale on artwork is present on focus.
3. `window.CONTEXT.setTheme({componentConfig:{Tile:{artwork:{style:{imageScale:5,},},style:{paddingYProgress:10,paddingYBetweenContent:10,},},},});`
4. After update when focused you should see an imageScale effect when focusing on the Tile
5. Any combination not including the scale property should clear it. `window.CONTEXT.setTheme({componentConfig:{Tile:{artwork:{},style:{paddingYProgress:10,paddingYBetweenContent:10,},},},});`
6. Setting any value on an instance of the component itself should not allow the componentConfig to overwrite it. 

![image](https://github.com/rdkcentral/Lightning-UI-Components/assets/2578683/8b72255a-dda0-4fe9-8bef-95278a3d43f1)



## Automation



## Checklist
